src/libcprover-cpp/api.h

```diff
@@ -20,6 +19,34 @@ class message_handlert;
 
 #include "options.h" // IWYU pragma: keep
 
+/// Opaque message type. Properties of messages to be fetched through further
+/// api calls.
+struct api_messaget;
+
+/// Given a \p api_message, this function returns that message expressed as a
+/// C language string.
+/// \note The memory for the returned string is owned by the message and as such
+/// does not need to be freed by users of the API.
+const char *api_message_get_string(const api_messaget &message);
+
+/// The type of pointers to contextual data passed to the api_message_callback
+/// functions. These pointers point to api consumer data and are just passed
+/// through to the callback verbatim. These support users of the api to avoid
+/// using global variables.
+using api_call_back_contextt = void *;
+
+/// The type of call back for feedback of status information and results.
+/// \param message: A structured message object. The lifetime of this object is
```

Comment:
currently this is just a string, right?

Reply:
Currently, yes. The idea is that this data structure would be extended on the opaque side with additional getters added on the public API side to fetch the structured data. That way we can extend the API without breaking any existing API consumers.